salt/metalk8s/map.jinja

```diff
@@ -2,38 +2,77 @@
 {% import_json 'metalk8s/versions.json' as versions with context %}
 
 {# First merge the basic defaults with versions information #}
-{% set version_defaults = salt['grains.filter_by']({
-  'CentOS': {
-    'kubernetes': versions.kubernetes,
-    'repo': {
-      'packages': versions.packages.centos,
-      'images': versions.images,
+{% set version_defaults = salt['grains.filter_by'](
+  salt['grains.filter_by']({
+    'CentOS': {
+      '7': {
+        'kubernetes': versions.kubernetes,
+        'repo': {
+          'packages': versions.packages.centos['7'],
+          'images': versions.images,
+        },
+      },
+      '8': {
+        'kubernetes': versions.kubernetes,
+        'repo': {
+          'packages': versions.packages.centos['8'],
+          'images': versions.images,
+        },
+      },
     },
-  },
-  'RedHat': {
-    'kubernetes': versions.kubernetes,
-    'repo': {
-      'packages': versions.packages.redhat,
-      'images': versions.images,
+    'RedHat': {
+      '7': {
+        'kubernetes': versions.kubernetes,
+        'repo': {
+          'packages': versions.packages.redhat['7'],
+          'images': versions.images,
+        },
+      },
+      '8': {
+        'kubernetes': versions.kubernetes,
+        'repo': {
+          'packages': versions.packages.redhat['8'],
+          'images': versions.images,
+        },
+      },
     },
-  },
-  'Ubuntu': {
-    'kubernetes': versions.kubernetes,
-    'repo': {
-      'packages': versions.packages.ubuntu,
-      'images': versions.images,
+    'Ubuntu': {
+      '18.04': {
+        'kubernetes': versions.kubernetes,
+        'repo': {
+          'packages': versions.packages.ubuntu['18.04'],
+          'images': versions.images,
+        },
+      },
     },
-  },
-}, grain='os', merge=defaults.get('version_defaults')) %}
+  }, grain='os'),
+  grain='osmajorrelease', merge=defaults.get('version_defaults'),
+) %}
 
-{% set packages = salt['grains.filter_by']({
-  'RedHat': {},
-  'Debian': {
-    'iproute': 'iproute2',
-    'm2crypto': 'python-m2crypto',
-    'python36-pyOpenSSL': 'python3-openssl',
-  },
-}, grain='os_family') %}
+{% set packages = salt['grains.filter_by'](
+  salt['grains.filter_by']({
+    'RedHat': {
+      '7': {
+        'python3': 'python36-rpm',
```

Comment:
This is a bit confusing to me.

Reply:
Yes, it is really confusing, because it's a mistake. :)